string/TextCursor.test.ts

```diff
@@ -0,0 +1,223 @@
+    assertEquals(cursor.move(7).location, { offset: 7, line: 4, column: 2 });
+    assertEquals(cursor.move(8).location, { offset: 8, line: 5, column: 1 });
+    assertEquals(cursor.move(9).location, { offset: 9, line: 5, column: 2 });
+  });
+});
+
+describe("TextCursor.line", () => {
+  it("is the line at the index", () => {
+    const text = "a\n\nbc\nd\nef";
+
+    assertEquals(new TextCursor(text).line, "a\n");
+    assertEquals(new TextCursor(text, 1).line, "a\n");
+    assertEquals(new TextCursor(text, 2).line, "\n");
+    assertEquals(new TextCursor(text, 3).line, "bc\n");
+    assertEquals(new TextCursor(text, 4).line, "bc\n");
+    assertEquals(new TextCursor(text, 5).line, "bc\n");
+    assertEquals(new TextCursor(text, 6).line, "d\n");
+    assertEquals(new TextCursor(text, 7).line, "d\n");
+    assertEquals(new TextCursor(text, 8).line, "ef");
+    assertEquals(new TextCursor(text, 9).line, "ef");
+  });
+});
+
+describe("TextCursor.toString()", () => {
+  it("is elided around index", () =>
+    assertEquals(
+      new TextCursor("abcdefghijklmnopqrstuvwxyz", 13).toString(),
+      'TextCursor("…efghijklmnopqrstuv…", 13)',
+    ));
+});
+
+describe("TextCursor.startsWith()", () => {
+  it("matches from the index", () => {
+    assert(new TextCursor(str).startsWith("feat"));
+    assert(!new TextCursor(str).startsWith("wxyz"));
+    assert(new TextCursor(str, 4).startsWith(": ex"));
+  });
+});
+
+describe("inspect", () => {
+  it("depicts the location", () =>
+    assertEquals(
+      stripColor(new TextCursor(str, 3).inspect()),
+      dedent(`
+        feat: example
+           ^
+      `).trim(),
+    ));
+
+  it("elides long lines", () =>
+    assertEquals(
+      stripColor(new TextCursor(longStr, 37).inspect({ maxLength })),
+      dedent(`
+        … jumps over the lazy dog then does it…
+                           ^
+      `).trim(),
+    ));
+
+  describe("line numbers", () => {
+    it("shows when >1 lines", () => {
+      const cursor = new TextCursor("a\nb", 1);
+
+      assertEquals(
+        stripColor(cursor.inspect()),
+        dedent(`
+          [L1] a¶
+                ^
+        `).trim(),
+      );
+
+      assertEquals(
+        stripColor(new TextCursor("a\nb", 2).inspect()),
+        dedent(`
+          [L2] b
+               ^
+        `).trim(),
+      );
+    });
+
+    it("is configurable", () => {
+      const cursor = new TextCursor("a\nb", 1);
+
+      assertEquals(
+        stripColor(cursor.inspect({ lineNumber: false })),
+        dedent(`
+          a¶
+           ^
+        `).trim(),
+      );
+
+      assertEquals(
+        stripColor(cursor.move(1).inspect({ lineNumber: false })),
+        dedent(`
+          b
+          ^
+        `).trim(),
+      );
+    });
+  });
+
+  describe("max length", () => {
+    it("defaults to 40 or console width", () => {
+      // TODO: Consider refactoring to allow stubbing console width.
```

Comment:
Did you want to leave this in?

Reply:
Yep, this needs some additional plumbing that may not be worth the effort. It is intentional!